roles/common/tasks/populate-asm-disks.yml

```diff
@@ -1,62 +1,43 @@
-- name: Collect new disk facts
-  setup:
-    gather_subset:
-      - '!all'
-      - '!any'
-      - devices
+- name: Resolve blk_device paths to first partition paths if they exist
+  ansible.builtin.shell: >
+    set -o pipefail;
+    case "{{ item.blk_device }}" in
+      /dev/disk/by-id/*)
+        part_path="{{ item.blk_device }}"-part1;
+        ;;
+      /dev/mapper/*)
+        part_path="{{ item.blk_device }}"p1;
```

Comment:
Not necessarily. Those devices are likely logical volumes managed by LVM and they don't need partitioning like traditional block devices. Since they are abstract volumes, they can be resized or relocated without the need to repartition the physical disks. Therefore, we don't need to track their partition paths.
I've updated the code to skip creating first_partition_id field for devices in /dev/mapper folder. PTAL ?

Reply:
I'd say sorta-kinda;  although it has been a convention here in the past, the presence of "mapper" in the device name doesn't necessarily indicate a LVM device.  LVM devices can be referenced as /dev/vg_name, and device-mapper can abstract non-LVM devices.  In the bms-toolkit case, actually the most commoon case is for persistent binding of SAN volumes by WWID.

It brings us back to the original question:  when should we, or should we not, partition disks.  I personally hate disk partitions:  they add complexity, risk I/O misalignment and I/O amplification, complicate resizing, and add the reuqirement for a partiion scan before mounting.  However they do add a form of a platform-independent disk label, and most Oracle shops do use them.  And, officially, Oracle recommends their use for ASM: https://docs.oracle.com/en/database/oracle/oracle-database/19/ladbi/creating-das-or-san-disk-partitions-for-oracle-automatic-storage-management.html.  It's worth noting that they recommend only a single whole-disk partition, so it's not partitioning as in traditional sense, but just a way of labelling the disk.

As I'm sure you've seen in creating the testscases though, it's time-consuming and error-prone to have so many different cases.  What do you think about either always partitioning or never partitioning, for consistency?